util/src/test/java/io/kubernetes/client/ExecTest.java

```diff
@@ -110,14 +113,17 @@ public void testExecProcess() throws IOException, InterruptedException {
 
     final ByteArrayOutputStream stdout = new ByteArrayOutputStream();
     final ByteArrayOutputStream stderr = new ByteArrayOutputStream();
-
-    Thread t1 = asyncCopy(process.getInputStream(), stdout);
-    Thread t2 = asyncCopy(process.getErrorStream(), stderr);
+    final Semaphore sem = new Semaphore(2);
+    sem.acquire(2);
+    Thread t1 = asyncCopy(process.getInputStream(), stdout, sem);
+    Thread t2 = asyncCopy(process.getErrorStream(), stderr, sem);
 
     process.getHandler().bytesMessage(makeStream(3, OUTPUT_EXIT0.getBytes(StandardCharsets.UTF_8)));
-    // TODO: Fix this asap!
-    Thread.sleep(1000);
 
+    while (!sem.tryAcquire(2)) {
```

Comment:
As above, why not just `sem.acquire(2)`?

Reply:
wanted to print message on why we are waiting..
but i have changed semaphores to countdown latch, as i thought it would be better than sempahore in these situations